Update the size of the memory allocation chunk in case of TizenRT

rtakacs opened the pull request:
Since #58 is landed (enable release build for TizenRT), the size of the allocation chunk is changed from 16 bytes to 8 bytes:

In `TizenRT/os/include/tinyara/mm/mm.h`:

```c
#ifdef CONFIG_DEBUG_MM_HEAPINFO
/* 16 = (uint32_t + uint32_t + uint32_t + uint16_t + uint16_t ) */
#define SIZEOF_MM_ALLOCNODE  (sizeof(mmsize_t) + sizeof(mmsize_t) + SIZEOF_MM_MALLOC_DEBUG_INFO)
#else
/* 8 = (uint32_t + uint32_t) */
#define SIZEOF_MM_ALLOCNODE   (sizeof(mmsize_t) + sizeof(mmsize_t))
#endif
```

@hs0225 This patch should fix the invalid big memory values at the [artik testpage](https://samsung.github.io/js-remote-test/#).

glistening commented:
@rtakacs 
i) What about including the explanation (what the magic number 8 or 16 means) in patch?
ii) According to `mm.h`, the size seemingly depends on whether CONFIG_DEBUG_MM_HEAPINFO is enabled or not, but in our patch, it is same 8. Do you assume that we will only use release build only for js-remote-test?
iii) After this PR, nuttx and tizenrt uses same code, why don't you merge these two condition?

rtakacs commented:
@glistening
i) Good idea, I've added a comment into the files.
ii) You are right, TizenRT uses different malloc header sizes (8 and 16 bytes) that should be supported for release and debug build. Fixed.
iii) I've merged the conditions.

yichoi commented:
LGTML
